example-2/evm/test/helpers/TestHelpers.sol

```diff
@@ -0,0 +1,678 @@
+        uint8 wormholeFinality = 1;
+        uint256 interestAccrualIndexPrecision = 10 ** 6;
+        uint256 collateralizationRatioPrecision = 10 ** 6;
+        uint256 maxLiquidationBonus = 125 * 10**4;
+        uint256 maxLiquidationPortion = 50;
+        uint256 maxLiquidationPortionPrecision = 10 ** 2;
+        uint8 oracleMode = 1;
+        uint64 priceStandardDeviations = 424;
+        uint64 priceStandardDeviationsPrecision = 10 ** 2;
+        address pythAddress = vm.envAddress("TESTING_PYTH_ADDRESS_AVAX");
+
+        Hub hub = new Hub(
+            address(wormholeContract), 
+            address(tokenBridgeContract), 
+            wormholeFinality, 
+            pythAddress, 
+            oracleMode,
+            priceStandardDeviations,
+            priceStandardDeviationsPrecision, 
+            maxLiquidationBonus, 
+            maxLiquidationPortion, 
+            maxLiquidationPortionPrecision,
+            interestAccrualIndexPrecision, 
+            collateralizationRatioPrecision
+        );
+
+        setOracleMode(oracleMode);
+
+        setHubData(HubData({
+            guardianSigner: guardianSigner,
+            wormholeSimulator: wormholeSimulator,
+            wormholeContract: wormholeContract,
+            tokenBridgeContract: tokenBridgeContract,
+            hub: hub,
+            hubChainId: wormholeContract.chainId()
+        }));
+
+        setVm(vm);
+
+        setPublishTime(1);
+        
+        registerChainOnHub(6, bytes32(uint256(uint160(vm.envAddress("TESTING_TOKEN_BRIDGE_ADDRESS_AVAX")))));
+
+    }
+
+    function doRegisterSpoke(uint256 index) internal returns (Spoke) {
+
+        SpokeData memory spokeData = getSpokeData(index);
+        // register asset
+        getHub().registerSpoke(
+            spokeData.foreignChainId, address(spokeData.spoke)
+        );
+
+        return spokeData.spoke;
+    }
+
+    function doRegisterAsset(Asset memory asset) internal {
+        Vm vm = getVm();
+        
+        uint256 reservePrecision = 1 * 10**6;
+        
+        // register asset
+        vm.recordLogs();
+        getHub().registerAsset(
+            asset.assetAddress, 
+            asset.collateralizationRatioDeposit, 
+            asset.collateralizationRatioBorrow,
+            asset.ratePrecision,
+            asset.kinks,
+            asset.rates,
+            asset.reserveFactor, 
+            reservePrecision, 
+            asset.pythId
+        );
+        
+        AssetInfo memory info = getHub().getAssetInfo(asset.assetAddress);
+
+        bool kinksMatch = true;
+        bool ratesMatch = true;
+
+        require(info.interestRateModel.kinks.length == asset.kinks.length, "lengths of kinks arrays don't match");
+        require(info.interestRateModel.rates.length == asset.rates.length, "lengths of rates arrays don't match");
+
+        for(uint i=0; i < asset.kinks.length; i++) {
+            if (info.interestRateModel.kinks[i] != asset.kinks[i]) {
+                kinksMatch = false;
```

Comment:
Can bail out early here with `revert` or `require(info.interestRateModel.kinks[i] != asset.kinks[i])` since if this is `false` it will revert at the end anyway.